test/e2e/queue.go

```diff
@@ -19,14 +19,187 @@ package e2e
 	"context"
 	"fmt"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strconv"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	busv1alpha1 "volcano.sh/volcano/pkg/apis/bus/v1alpha1"
+	schedulingv1beta1 "volcano.sh/volcano/pkg/apis/scheduling/v1beta1"
+	"volcano.sh/volcano/pkg/cli/util"
 )
 
 var _ = Describe("Queue E2E Test", func() {
+	It("Queue Command Close And Open With State Check", func() {
+		q1 := "queue-command-close"
+		defaultQueue := "default"
+		ctx := initTestContext(options{
+			queues: []string{q1},
+		})
+		defer cleanupTestContext(ctx)
+
+		By("Close queue command check")
+		err := util.CreateQueueCommand(ctx.vcclient, defaultQueue, q1, busv1alpha1.CloseQueueAction)
+		if err != nil {
+			Expect(err).NotTo(HaveOccurred(), "Error send close queue command")
+		}
+
+		err = waitQueueStatus(func() (bool, error) {
+			queue, err := ctx.vcclient.SchedulingV1beta1().Queues().Get(context.TODO(), q1, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+			return queue.Status.State == schedulingv1beta1.QueueStateClosed, nil
+		})
+		Expect(err).NotTo(HaveOccurred(), "Wait for closed queue %s failed", q1)
+
+		By("Open queue command check")
+		err = util.CreateQueueCommand(ctx.vcclient, defaultQueue, q1, busv1alpha1.OpenQueueAction)
+		if err != nil {
+			Expect(err).NotTo(HaveOccurred(), "Error send open queue command")
+		}
+
+		err = waitQueueStatus(func() (bool, error) {
+			queue, err := ctx.vcclient.SchedulingV1beta1().Queues().Get(context.TODO(), q1, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Get queue %s failed", q1)
+			return queue.Status.State == schedulingv1beta1.QueueStateOpen, nil
+		})
+		Expect(err).NotTo(HaveOccurred(), "Wait for reopen queue %s failed", q1)
+
+	})
+
+	Describe("Queue Job Status Transaction", func() {
+		var ctx *testContext
+		var q1 string
+		var podNamespace string
+		var rep int32
+		var firstJobName string
+		By("Prepare 3 job")
+		BeforeEach(func() {
+
+			q1 = "queue-jobs-status-transaction"
+			ctx = initTestContext(options{
+				queues: []string{q1},
+			})
+			podNamespace = ctx.namespace
+			slot := halfCPU
+			rep = clusterSize(ctx, slot)
+
+			if rep < 4 {
+				err := fmt.Errorf("You need at least 2 logical cpu for this test case, please skip 'Queue Job Status Transaction' when you see this message")
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			for i := 0; i < 3; i++ {
+				spec := &jobSpec{
+					tasks: []taskSpec{
+						{
+							img: defaultNginxImage,
+							req: slot,
+							min: rep,
+							rep: rep,
+						},
+					},
+				}
+				spec.name = "queue-job-status-transaction-test-job-" + strconv.Itoa(i)
+				if i == 0 {
+					firstJobName = spec.name
+				}
+				spec.queue = q1
+				createJob(ctx, spec)
+			}
+		})
+
+		AfterEach(func() {
+			cleanupTestContext(ctx)
+		})
+
+		Context("When transform from Inqueue to Running", func() {
```

Comment:
I donot think Context is necessary here, right?

Reply:
Not necessary, but better use it in one behavior different circumstances.